tests/storage/restore.go

```diff
@@ -168,39 +169,28 @@ var _ = SIGDescribe("[Serial]VirtualMachineRestore Tests", func() {
 		return pvc
 	}
 
-	waitDeleted := func(deleteFunc func() error) {
-		Eventually(func() bool {
-			err := deleteFunc()
-			if errors.IsNotFound(err) {
-				return true
-			}
-			Expect(err).ToNot(HaveOccurred())
-			return false
-		}, 180*time.Second, time.Second).Should(BeTrue())
-	}
-
 	deleteVM := func(vm *v1.VirtualMachine) {
-		waitDeleted(func() error {
-			return virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})
-		})
+		err := virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})
+		noDeleteErr := err == nil || errors.IsNotFound(err)
```

Comment:
```
if errors.IsNotFound(err){
    err = nil
}
Expect(err).NotTo(HaveOccured())
```
Please use this so we have a better picture of what went wrong.

Reply:
Yep way better, done